ntoskrnl/mm/ARM3/miarm.h

```diff
@@ -1002,7 +1002,8 @@ MI_WRITE_INVALID_PTE(IN PMMPTE PointerPte,
 {
     /* Write the invalid PTE */
     ASSERT(InvalidPte.u.Hard.Valid == 0);
-    ASSERT(InvalidPte.u.Long != 0);
+    /* Added by Timo Kreuzer in https://github.com/reactos/reactos/commit/4019985 */
+//    ASSERT(InvalidPte.u.Long != 0);
```

Comment:
Please provide a backtrace of when this is triggered

Reply:
Here is the backtrace as requested:
```
[ROSAUTOTEST] System uptime 49.39 seconds
0364:0368:Could not find dependent assembly Microsoft.Windows.Common-Controls (5.82.4704.0)
0364:0368:Failed to create the implicit act ctx. Status: 0xc0150002!!!
(ntoskrnl\mm\ARM3\sysldr.c:170) Loading: \??\C:\ReactOS\bin\kmtest_drv.sys at F7945000 with 1e1 pages
Running Wine Test, Module: kmtest, Test: MmReservedMapping
0378:037c:Could not find dependent assembly Microsoft.Windows.Common-Controls (5.82.4704.0)
0378:037c:Failed to create the implicit act ctx. Status: 0xc0150002!!!
err:(dll\win32\advapi32\service\scm.c:3003) RStartServiceW() failed (Error 1056)
(modules\rostests\kmtests\kmtest_drv\kmtest_drv.c:371) DriverIoControl. Starting test MmReservedMapping
(ntoskrnl\mm\freelist.c:196) ARM3-DEBUG: Being called with 0 ffffffffffffffff 0 1000 1 0
(ntoskrnl\mm\freelist.c:196) ARM3-DEBUG: Being called with 0 ffffffffffffffff 0 a000 1 0
(ntoskrnl\mm\freelist.c:196) ARM3-DEBUG: Being called with 0 ffffffffffffffff 0 b000 1 0
Assertion d:\reacto\reactos\ntoskrnl\mm\arm3\miarm.h(1049): InvalidPde.u.Long != 0
nt!MI_WRITE_INVALID_PTE+0x65:
804d5135 cd2c            int     2Ch
kd> k
 # ChildEBP RetAddr  
00 f7b449ec 804dfc51 nt!MI_WRITE_INVALID_PTE+0x65 [d:\reacto\reactos\ntoskrnl\mm\arm3\miarm.h @ 1049]
01 f7b44a28 f79ddaa6 nt!MmAllocateMappingAddress+0x191 [d:\reacto\reactos\ntoskrnl\mm\arm3\pool.c @ 1615]
02 f7b44a74 f7946820 kmtest_drv!Test_MmReservedMapping+0x276 [d:\reacto\reactos\modules\rostests\kmtests\ntos_mm\mmreservedmapping.c @ 326]
03 f7b44b3c 80488eb7 kmtest_drv!DriverIoControl+0x310 [d:\reacto\reactos\modules\rostests\kmtests\kmtest_drv\kmtest_drv.c @ 372]
04 f7b44b68 804807ec nt!IofCallDriver+0xc7 [d:\reacto\reactos\ntoskrnl\io\iomgr\irp.c @ 1286]
05 f7b44ba4 8047fb9d nt!IopPerformSynchronousRequest+0x7c [d:\reacto\reactos\ntoskrnl\io\iomgr\iofunc.c @ 142]
06 f7b44ca4 80480df4 nt!IopDeviceFsIoControl+0x8ad [d:\reacto\reactos\ntoskrnl\io\iomgr\iofunc.c @ 648]
07 f7b44cd8 8040405e nt!NtDeviceIoControlFile+0x34 [d:\reacto\reactos\ntoskrnl\io\iomgr\iofunc.c @ 1453]
08 f7b44d10 8056bbd8 nt!KiSystemCallTrampoline+0x19
09 f7b44d5c 80403ea5 nt!KiSystemServiceHandler+0x278 [d:\reacto\reactos\ntoskrnl\ke\i386\traphdlr.c @ 1840]
0a f7b44d5c 7c935ffe nt!KiFastCallEntry+0x96
0b 0012fdc0 7c96ca67 ntdll!KiFastSystemCallRet
0c 0012fdc4 7c67d869 ntdll!NtDeviceIoControlFile+0xc
0d 0012fe3c 00404010 kernel32!DeviceIoControl+0x1e9 [d:\reacto\reactos\dll\win32\kernel32\client\file\deviceio.c @ 244]
0e 0012fe7c 00402d4c kmtest_!KmtRunKernelTest+0x70 [d:\reacto\reactos\modules\rostests\kmtests\kmtest\support.c @ 104]
0f 0012fea0 0040305a kmtest_!RunTest+0x10c [d:\reacto\reactos\modules\rostests\kmtests\kmtest\kmtest.c @ 281]
10 0012fecc 0041e78a kmtest_!main+0x2aa [d:\reacto\reactos\modules\rostests\kmtests\kmtest\kmtest.c @ 366]
11 0012ffb4 0041ea68 kmtest_!__tmainCRTStartup+0x2ba [d:\reacto\reactos\sdk\lib\crt\startup\crtexe.c @ 311]
12 0012ffc0 7c656f34 kmtest_!mainCRTStartup+0x28 [d:\reacto\reactos\sdk\lib\crt\startup\crtexe.c @ 192]
13 0012fff0 00000000 kernel32!BaseProcessStartup+0x54 [d:\reacto\reactos\dll\win32\kernel32\client\proc.c @ 463]
kd> gh
(ntoskrnl\mm\freelist.c:196) ARM3-DEBUG: Being called with 0 ffffffffffffffff 0 1000 1 0
(ntoskrnl\mm\freelist.c:196) ARM3-DEBUG: Being called with 0 ffffffffffffffff 0 1000 1 0
(ntoskrnl\mm\freelist.c:196) ARM3-DEBUG: Being called with 0 ffffffffffffffff 0 2000 1 0
(ntoskrnl\mm\freelist.c:196) ARM3-DEBUG: Being called with 0 ffffffffffffffff 0 1000 1 0
(ntoskrnl\mm\freelist.c:196) ARM3-DEBUG: Being called with 0 ffffffffffffffff 0 1000 1 0
(ntoskrnl\mm\freelist.c:196) ARM3-DEBUG: Being called with 0 ffffffffffffffff 0 2000 1 0
(modules\rostests\kmtests\kmtest_drv\kmtest_drv.c:373) DriverIoControl. Finished test MmReservedMapping
MmReservedMapping: 213 tests executed (0 marked as todo, 0 failures), 0 skipped.
Test MmReservedMapping completed in 0.08 seconds.


SYSREG_CHECKPOINT:THIRDBOOT_COMPLETE
```

Reply:
This seems to be another assert. You disabled it on line 1005, while in backtrace is another one on line 1049. However I don't see any condition in the code which uses `MI_WRITE_INVALID_PDE` instead of `MI_WRITE_INVALID_PTE`. Or it's redirected somewhere else? :thinking:

Reply:
These two lines have been completely removed now and the crash/hang is no longer occurring with the latest code.